test/src/test/java/hudson/model/FingerprintTest.java

```diff
@@ -413,6 +414,24 @@ public void checkNormalFingerprintWithWebClient() throws Exception {
         assertEquals(200, page2.getWebResponse().getStatusCode());
     }
 
+    @Test
+    @Issue("JENKINS-65611")
+    public void canModifyFacets() {
+        Fingerprint fingerprint = new Fingerprint(new Fingerprint.BuildPtr("foo", 3),
+                "stuff&more.jar", Util.fromHexString(SOME_MD5));
+        TestFacet testFacet = new TestFacet(fingerprint, 0, "test");
+        assertThat(fingerprint.getFacets().size(), is(0));
+        fingerprint.getFacets().add(testFacet);
+        assertThat(fingerprint.getFacets().size(), is(1));
+        assertTrue(fingerprint.getFacets().contains(testFacet));
+        fingerprint.getFacets().remove(testFacet);
+        assertThat(fingerprint.getFacets().size(), is(0));
+        fingerprint.getFacets().add(testFacet);
+        Iterator<FingerprintFacet> itr = fingerprint.getFacets().iterator();
+        itr.next();
+        itr.remove();
```

Comment:
I think you want to actually do a test here to check the thing is actually removed as opposed to silently fails.

Looking at the code I am not convinced that this is anything other than a no-op

https://github.com/jenkinsci/jenkins/blob/a1175b932e1ac97085d69752f5821141d012314f/core/src/main/java/hudson/util/Iterators.java#L337-L344

Reply:
That function is a little misleading it is actually https://github.com/jenkinsci/jenkins/blob/a1175b932e1ac97085d69752f5821141d012314f/core/src/main/java/hudson/util/Iterators.java#L367-L369